File: skill_sdk/tests/fixtures/ts-project/src/core/ProcessorConfig.ts
/**
 * Configuration for processors. Mirrors ProcessorConfig in the Java fixture.
 */
export class ProcessorConfig {
  constructor(
    private readonly timeout: number,
    private readonly retries: number
  ) {}

  getTimeout(): number {
    return this.timeout;
  }

  getRetries(): number {
    return this.retries;
  }
}

File: skill_sdk/tests/fixtures/rust-project/src/core/processor_config.rs
/// ProcessorConfig holds configuration for data processors.
#[derive(Debug, Clone)]
pub struct ProcessorConfig {
    pub timeout: u32,
    pub retries: u32,
}

impl Default for ProcessorConfig {
    fn default() -> Self {
        Self {
            timeout: 30,
            retries: 3,
        }
    }
}
